diff --git a/src/System/MarDirectorEvent.cpp b/src/System/MarDirectorEvent.cpp
index a847f338..f4e00287 100644
--- a/src/System/MarDirectorEvent.cpp
+++ b/src/System/MarDirectorEvent.cpp
@@ -245,8 +245,8 @@ void TMarDirector::fireStreamingMovie(u8 param_1)
 {
 	switch (param_1) {
 	case 0:
-		if (!(unk4C & 0x100)) {
-			unk4C |= 0x100;
+		if (!((*(volatile u16*)&unk4C) & 0x100)) {
+			(*(volatile u16*)&unk4C) |= 0x100;
 			setNextStage(0x1, nullptr);
 			TFlagManager::smInstance->setBool(true, 0x10389);
 			TFlagManager::smInstance->setBool(true, 0x30004);
@@ -255,49 +255,49 @@ void TMarDirector::fireStreamingMovie(u8 param_1)
 		break;
 
 	case 10:
-	case 12:
-		if (!(unk4C & 0x100)) {
-			unk4C |= 0x100;
+		if (!((*(volatile u16*)&unk4C) & 0x100)) {
+			(*(volatile u16*)&unk4C) |= 0x100;
 			setNextStage(0x3B, nullptr);
 			gpApplication.mMovie = param_1;
 		}
 		break;
 
 	case 7:
-		if (!(unk4C & 0x100)) {
-			unk4C |= 0x100;
+		if (!((*(volatile u16*)&unk4C) & 0x100)) {
+			(*(volatile u16*)&unk4C) |= 0x100;
 			setNextStage(0xE06, nullptr);
 			gpApplication.mMovie = param_1;
 		}
 		break;
 
 	case 8:
-		if (!(unk4C & 0x100)) {
-			unk4C |= 0x100;
+		if (!((*(volatile u16*)&unk4C) & 0x100)) {
+			(*(volatile u16*)&unk4C) |= 0x100;
 			setNextStage(0xE07, nullptr);
 			gpApplication.mMovie = param_1;
 		}
 		break;
 
 	case 11:
-		if (!(unk4C & 0x100)) {
-			unk4C |= 0x100;
+		if (!((*(volatile u16*)&unk4C) & 0x100)) {
+			(*(volatile u16*)&unk4C) |= 0x100;
 			setNextStage(0x3C, nullptr);
 			gpApplication.mMovie = param_1;
 		}
 		break;
 
 	case 2:
-		if (!(unk4C & 0x100)) {
-			unk4C |= 0x100;
+		if (!((*(volatile u16*)&unk4C) & 0x100)) {
+			(*(volatile u16*)&unk4C) |= 0x100;
 			setNextStage(0x101, nullptr);
 			gpApplication.mMovie = param_1;
 		}
 		break;
 
+	case 12:
 	default:
-		if (!(unk4C & 0x100)) {
-			unk4C |= 0x100;
+		if (!((*(volatile u16*)&unk4C) & 0x100)) {
+			(*(volatile u16*)&unk4C) |= 0x100;
 			setNextStage(0xF, nullptr);
 			gpApplication.mMovie = (u8)param_1;
 		}
