diff --git a/src/System/MarNameRefGen_BossEnemy.cpp b/src/System/MarNameRefGen_BossEnemy.cpp
index bf24c803..6945561a 100644
--- a/src/System/MarNameRefGen_BossEnemy.cpp
+++ b/src/System/MarNameRefGen_BossEnemy.cpp
@@ -1,3 +1,91 @@
+static const char* dummyMactorStringValue1 = "\0\0\0\0\0\0\0\0\0\0\0";
+static const char* SMS_NO_MEMORY_MESSAGE   = "メモリが足りません\n";
+
+static const char cDirtyFileName[] = "/scene/map/pollution/H_ma_rak.bti";
+static const char cDirtyTexName[]  = "H_ma_rak_dummy";
+
+static const char* MtxCalcTypeName[] = {
+	"MActorMtxCalcType_Basic クラシックスケールＯＮ",
+	"MActorMtxCalcType_Softimage クラシックスケールＯＦＦ",
+	"MActorMtxCalcType_MotionBlend モーションブレンド",
+	"MActorMtxCalcType_User ユーザー定義",
+};
+
+static const char* bossEnemyNames[] = {
+	"マリオモドキ",
+	"EMarioManager",
+	"典型敵マネージャ",
+	"BossHanachan",
+	"BossHanachanManager",
+	"SleepBossHanachan",
+	"SleepBossHanachanManager",
+	"/enemy/sleepBossHanachan.prm",
+	"BossEelManager",
+	"BEelTearsManager",
+	"めおとウナギ涙マネージャー",
+	"KoopaManager",
+	"クッパマネージャー",
+	"HinoKuri2",
+	"ヒノクリ２",
+	"HinoKuri2Manager",
+	"ヒノクリ２マネージャ",
+	"BossGesso",
+	"ボスゲッソー",
+	"BossGessoManager",
+	"ボスゲッソーマネージャ",
+	"TinKoopa",
+	"メカクッパ",
+	"TinKoopaManager",
+	"メカクッパマネージャ",
+	"CoasterKillerManager",
+	"コースターキラーマネージャー",
+	"CoasterKiller",
+	"コースターキラー",
+	"KoopaJrManager",
+	"クッパジュニアマネージャー",
+	"クッパジュニア",
+	"KoopaJrSubmarineManager",
+	"クッパジュニアサブマリンマネージャー",
+	"KoopaJrSubmarine",
+	"クッパジュニアサブマリン",
+	"LimitKoopaJrManager",
+	"リミットクッパジュニアマネージャー",
+	"LimitKoopaJr",
+	"リミットクッパジュニア",
+	"LimitKoopaManager",
+	"LimitKoopa",
+	"BathtubKillerManager",
+	"バスタブキラーマネージャー",
+	"BathtubKiller",
+	"バスタブキラー",
+	"BathtubPeachManager",
+	"バスタブピーチマネージャー",
+	"BathtubPeach",
+	"バスタブピーチ",
+	"BossWanwan",
+	"ボスワンワン",
+	"BossWanwanManager",
+	"ボスワンワンマネージャ",
+	"BossPakkun",
+	"ボスパックン改",
+	"KBossPakkun",
+	"ボスパックン軽",
+	"BossPakkunManager",
+	"ボスパックンマネージャー",
+	"KBossPakkunManager",
+	"ボスパックン軽マネージャ",
+	"BossTelesa",
+	"ボステレサ",
+	"BossTelesaManager",
+	"ボステレサマネージャー",
+	"BubbleManager",
+	"バブルマネージャー",
+	"BossManta",
+	"ボスマンタ",
+	"BossMantaManager",
+	"ボスマンタマネージャ",
+};
+
 #include "Enemy/BathtubKiller.hpp"
 #include "Enemy/BossEel.hpp"
 #include "Enemy/BossGesso.hpp"
