diff --git a/src/System/MarDirectorEvent.cpp b/src/System/MarDirectorEvent.cpp
index 3a6e01d6..a847f338 100644
--- a/src/System/MarDirectorEvent.cpp
+++ b/src/System/MarDirectorEvent.cpp
@@ -27,7 +27,10 @@ void isNearMapObj(const JDrama::TActor&, const char*, f32) { }
 
 void is3BarrelNear(const TBaseNPC&) { }
 
-void TMarDirector::getTalkMsgID(TBaseNPC*) { }
+void TMarDirector::getTalkMsgID(TBaseNPC*)
+{
+	(void)"\x83\x6a\x83\x52\x83\x7d\x83\x7d";
+}
 
 void TMarDirector::updateFlag(TBaseNPC*, u32, u32) { }
 
